### PR TITLE
test.apiv2: add test cases for committing an image from a container

### PR DESCRIPTION
**Pull request by chuanchang**
Testing query parameters: container, repo, tag, comment, author, pause.

```
ok 1 [20-containers] GET libpod/containers/json (at start: clean slate) : status=200
ok 2 [20-containers] GET libpod/containers/json (at start: clean slate) : length=0
ok 3 [20-containers] GET libpod/containers/json : status=200
ok 4 [20-containers] GET libpod/containers/json : length=0
ok 5 [20-containers] GET libpod/containers/json?all=true : status=200
ok 6 [20-containers] GET libpod/containers/json?all=true : length=1
ok 7 [20-containers] GET libpod/containers/json?all=true : .[0].Id~[0-9a-f]\{12\}
ok 8 [20-containers] GET libpod/containers/json?all=true : .[0].Image=quay.io/libpod/alpine_labels:latest
ok 9 [20-containers] GET libpod/containers/json?all=true : .[0].Command[0]=true
ok 10 [20-containers] GET libpod/containers/json?all=true : .[0].State~\(exited\|stopped\)
ok 11 [20-containers] GET libpod/containers/json?all=true : .[0].ExitCode=0
ok 12 [20-containers] GET libpod/containers/json?all=true : .[0].IsInfra=false
ok 13 [20-containers] DELETE libpod/containers/35b5ea620e953e6132ec0f248e61c7136da20bf9c245c467ec0b6aeccadcbf10 : status=204
ok 14 [20-containers] GET libpod/containers/json?all=true : status=200
ok 15 [20-containers] POST libpod/commit?container=nonesuch [] : status=404
ok 16 [20-containers] POST libpod/commit?container=myfoo&repo=newrepo&comment=foo&author=bob [] : status=500
ok 17 [20-containers] POST libpod/commit?container=myfoo [] : status=200
ok 18 [20-containers] GET libpod/images/43dee98fd21c79e14d3b610bd7e8d7efd0cccf01763b7eed2e6d606d2f391a9c/json : status=200
ok 19 [20-containers] GET libpod/images/43dee98fd21c79e14d3b610bd7e8d7efd0cccf01763b7eed2e6d606d2f391a9c/json : .RepoTags[0]=null
ok 20 [20-containers] GET libpod/images/43dee98fd21c79e14d3b610bd7e8d7efd0cccf01763b7eed2e6d606d2f391a9c/json : .Author=
ok 21 [20-containers] GET libpod/images/43dee98fd21c79e14d3b610bd7e8d7efd0cccf01763b7eed2e6d606d2f391a9c/json : .Comment=
ok 22 [20-containers] POST libpod/commit?container=myfoo&repo=newrepo&comment=foo&author=bob&format=docker [] : status=200
ok 23 [20-containers] GET libpod/images/newrepo:latest/json : status=200
ok 24 [20-containers] GET libpod/images/newrepo:latest/json : .RepoTags[0]=localhost/newrepo:latest
ok 25 [20-containers] GET libpod/images/newrepo:latest/json : .Author=bob
ok 26 [20-containers] GET libpod/images/newrepo:latest/json : .Comment=foo
ok 27 [20-containers] POST libpod/commit?container=453ccae856ebd4fd4de0f1238edb5d63bb1e9b378a913613f09eef07c9b844d2&repo=newrepo&tag=v1&author=alice&pause=false [] : status=200
ok 28 [20-containers] GET libpod/images/newrepo:v1/json : status=200
ok 29 [20-containers] GET libpod/images/newrepo:v1/json : .RepoTags[0]=localhost/newrepo:v1
ok 30 [20-containers] GET libpod/images/newrepo:v1/json : .Author=alice
ok 31 [20-containers] POST libpod/commit?container=453ccae856eb&repo=newrepo&tag=v2&comment=bar&author=eric&format=docker&pause=true [] : status=200
ok 32 [20-containers] GET libpod/images/newrepo:v2/json : status=200
ok 33 [20-containers] GET libpod/images/newrepo:v2/json : .RepoTags[0]=localhost/newrepo:v2
ok 34 [20-containers] GET libpod/images/newrepo:v2/json : .Author=eric
ok 35 [20-containers] GET libpod/images/newrepo:v2/json : .Comment=bar
ok 36 [20-containers] DELETE images/localhost/newrepo:latest?force=true : status=200
ok 37 [20-containers] DELETE images/localhost/newrepo:v1?force=true : status=200
ok 38 [20-containers] DELETE images/localhost/newrepo:v2?force=true : status=200
ok 39 [20-containers] DELETE libpod/containers/453ccae856ebd4fd4de0f1238edb5d63bb1e9b378a913613f09eef07c9b844d2 : status=204
1..39
```

Signed-off-by: Alex Jia <chuanchang.jia@gmail.com>

### CODE REVIEW

**Comment by baude** [comment]
@chuanchang thanks for your contributions!  Excellent!
LGTM

**Comment by mheon** [comment]
LGTM, but would like a final signoff from @edsantiago

File: test/apiv2/20-containers.at [hunk 1]
```diff
@@ -30,4 +30,52 @@ cid=$(jq -r '.[0].Id' <<<"$output")
...
 
+CNAME=myfoo
+podman run --name $CNAME $IMAGE -td /bin/sh
+t GET libpod/containers/json?all=true 200
+cid=$(jq -r '.[0].Id' <<<"$output")
+
+# No such container
+t POST "libpod/commit?container=nonesuch" '' 404
+
+# Only comment can be used with the docker format
```
Comment (edsantiago) [inline test/apiv2/20-containers.at:41]: Phrasing: try "Comment can only be used with docker format, not OCI'

**Comment by chuanchang** [comment]
> 
> 
> Nice work! One wording change request, and two places where you should do result validation. LGTM after that, thank you.

@edsantiago thank you, done, and also added testing for 'changes' query parameter, test result as below.

```
ok 1 [20-containers] GET libpod/containers/json (at start: clean slate) : status=200
ok 2 [20-containers] GET libpod/containers/json (at start: clean slate) : length=0
ok 3 [20-containers] GET libpod/containers/json : status=200
ok 4 [20-containers] GET libpod/containers/json : length=0
ok 5 [20-containers] GET libpod/containers/json?all=true : status=200
ok 6 [20-containers] GET libpod/containers/json?all=true : length=1
ok 7 [20-containers] GET libpod/containers/json?all=true : .[0].Id~[0-9a-f]\{12\}
ok 8 [20-containers] GET libpod/containers/json?all=true : .[0].Image=quay.io/libpod/alpine_labels:latest
ok 9 [20-containers] GET libpod/containers/json?all=true : .[0].Command[0]=true
ok 10 [20-containers] GET libpod/containers/json?all=true : .[0].State~\(exited\|stopped\)
ok 11 [20-containers] GET libpod/containers/json?all=true : .[0].ExitCode=0
ok 12 [20-containers] GET libpod/containers/json?all=true : .[0].IsInfra=false
ok 13 [20-containers] DELETE libpod/containers/0fd5f2f6afdd47d7be32f652831e5305fedd62bd3b18c5925d8777b2bf26d2e6 : status=204
ok 14 [20-containers] GET libpod/containers/json?all=true : status=200
ok 15 [20-containers] GET libpod/containers/json?all=true : .[0].Id~[0-9a-f]\{12\}
ok 16 [20-containers] POST libpod/commit?container=nonesuch [] : status=404
ok 17 [20-containers] POST libpod/commit?container=myfoo&repo=newrepo&comment=foo&author=bob [] : status=500
ok 18 [20-containers] POST libpod/commit?container=myfoo [] : status=200
ok 19 [20-containers] POST libpod/commit?container=myfoo [] : .Id~[0-9a-f]\{12\}
ok 20 [20-containers] GET libpod/images/20a38ea39f2c7b0d7c31261223fbdf36ccf9c39f1cf8b9c3643a7e184782f48c/json : status=200
ok 21 [20-containers] GET libpod/images/20a38ea39f2c7b0d7c31261223fbdf36ccf9c39f1cf8b9c3643a7e184782f48c/json : .RepoTags[0]=null
ok 22 [20-containers] GET libpod/images/20a38ea39f2c7b0d7c31261223fbdf36ccf9c39f1cf8b9c3643a7e184782f48c/json : .Author=
ok 23 [20-containers] GET libpod/images/20a38ea39f2c7b0d7c31261223fbdf36ccf9c39f1cf8b9c3643a7e184782f48c/json : .Comment=
ok 24 [20-containers] POST libpod/commit?container=myfoo&repo=newrepo&comment=foo&author=bob&format=docker [] : status=200
ok 25 [20-containers] GET libpod/images/newrepo:latest/json : status=200
ok 26 [20-containers] GET libpod/images/newrepo:latest/json : .RepoTags[0]=localhost/newrepo:latest
ok 27 [20-containers] GET libpod/images/newrepo:latest/json : .Author=bob
ok 28 [20-containers] GET libpod/images/newrepo:latest/json : .Comment=foo
ok 29 [20-containers] POST libpod/commit?container=037be41b4d9853e0e8d1a694319f3244c46ddd44bd85b3de2653f2d9b0c14d05&repo=newrepo&tag=v1&author=alice&pause=false [] : status=200
ok 30 [20-containers] GET libpod/images/newrepo:v1/json : status=200
ok 31 [20-containers] GET libpod/images/newrepo:v1/json : .RepoTags[0]=localhost/newrepo:v1
ok 32 [20-containers] GET libpod/images/newrepo:v1/json : .Author=alice
ok 33 [20-containers] POST libpod/commit?container=037be41b4d98&repo=newrepo&tag=v2&comment=bar&author=eric&format=docker&changes=CMD=/bin/foo&pause=true [] : status=200
ok 34 [20-containers] GET libpod/images/newrepo:v2/json : status=200
ok 35 [20-containers] GET libpod/images/newrepo:v2/json : .RepoTags[0]=localhost/newrepo:v2
ok 36 [20-containers] GET libpod/images/newrepo:v2/json : .Author=eric
ok 37 [20-containers] GET libpod/images/newrepo:v2/json : .Comment=bar
ok 38 [20-containers] GET libpod/images/newrepo:v2/json : .Config.Cmd[-1]=/bin/foo
ok 39 [20-containers] DELETE images/localhost/newrepo:latest?force=true : status=200
ok 40 [20-containers] DELETE images/localhost/newrepo:v1?force=true : status=200
ok 41 [20-containers] DELETE images/localhost/newrepo:v2?force=true : status=200
ok 42 [20-containers] DELETE libpod/containers/037be41b4d9853e0e8d1a694319f3244c46ddd44bd85b3de2653f2d9b0c14d05 : status=204
1..42
```

**Comment by edsantiago** [comment]
Nice work, thank you!

/lgtm

[Quick suggestion for next time: it is not necessary to paste in test results.]

**Pull request merged**